torch/_dynamo/symbolic_convert.py

```diff
@@ -245,12 +245,25 @@ def inner(self: "InstructionTranslatorBase", inst: Instruction):
                 self.jump(inst)
                 return
 
-            # Manually insert torch._assert instead of python assert and jump over
+            # Manually insert torch._assert_async instead of python assert and jump over
             # assert related instructions as we don't need them anymore.
+            scalar_to_tensor_proxy = self.output.create_proxy(
+                "call_function",
+                torch.scalar_tensor,
+                *proxy_args_kwargs((value,), {})
+            )
```

Comment:
Hmm, what if value is a TensorVariable?

Reply:
Good catch! If tensor, we just pass it into assert directly